src/providers/MeetingProvider/MeetingManager.ts

```diff
@@ -273,12 +272,13 @@ export class MeetingManager implements AudioVideoObserver {
 
     this.audioVideo = this.meetingSession.audioVideo;
     this.publishAudioVideo();
-    // When an attendee leaves, we remove AudioVideoObservers and nullify AudioVideoFacade and MeetingSession object.
-    // This results into missing few meeting events triggered with audioVideoDidStop such as meetingEnded, meetingFailed and meetingStartFailed.
+    // When an attendee leaves, we nullify AudioVideoFacade and MeetingSession object.
+    // We don't remove AudioVideoObservers in the leave function because AudioVideoObservers will be removed before being called.
     // We may also loose audioInputUnselected and videoInputUnselected events as we choose null devices when an attendee leaves.
-    // When a new AudioVideoFacade object is created remove and re-add the eventDidReceive observer which wont leak.
+    // When a new AudioVideoFacade object is created, remove and re-add the eventDidReceive observer which wont leak.
     this.audioVideo.removeObserver({ eventDidReceive: this.eventDidReceive });
     this.audioVideo.addObserver({ eventDidReceive: this.eventDidReceive });
+    this.audioVideo.removeObserver(this.audioVideoObservers);
```

Comment:
When do these observers get removed when we normally call meetingManager.leave()? Won't they still be active if you join a meeting and then leave a meeting? We'd never removeObservers because we never call initializeMeetingSession when we leave a meeting, right? Only when we join a meeting.

Reply:
Make sense, remove audio video observers in the `audioVideoDidStop()` function instead